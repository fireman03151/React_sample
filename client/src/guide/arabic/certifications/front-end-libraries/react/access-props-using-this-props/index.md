---
title: Access Props Using this.props
localeTitle: وصول الدعائم باستخدام هذا. المحاصيل
---
## \## Access Props باستخدام هذا .props

للحصول على هذا الحل ، يجب أن تتذكر أولاً كيفية إضافة الأدوات الدعائية إلى المكون الرئيسي:

\`\` \`JSX

 `Once you have the prop set, you can use this.props in your child component. 
` 

أتش تي أم أل

كلمة المرور المؤقتة هي: **{this.props.tempPassword}**

\`\` \`