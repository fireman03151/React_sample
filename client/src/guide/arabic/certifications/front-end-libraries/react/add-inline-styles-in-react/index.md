---
title: Add Inline Styles in React
localeTitle: إضافة أنماط مضمنة في React
---
## إضافة أنماط مضمنة في React

يمكنك تعريف نمط مكون تمرير الكائن مباشرة كـ "نمط" prop. فقط تذكر أن كل خاصية من كائن النمط هي camelcased. لذا ، تُعلن الخصائص مثل "font-size" بأنها "fontSize" لتكون خاصية كائن جافا سكريبت صالحة.

### المفسد

 `class Colorful extends React.Component { 
  render() { 
    // change code below this line 
    return ( 
      <div style={{color: "yellow", fontSize: 24}}>Style Me!</div> 
    ); 
    // change code above this line 
  } 
 }; 
` 

### مصادر

[DOM Elements Style](https://reactjs.org/docs/dom-elements.html#style)