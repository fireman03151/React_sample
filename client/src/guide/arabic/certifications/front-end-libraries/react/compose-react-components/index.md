---
title: Compose React Components
localeTitle: تأليف React Components
---
## تأليف React Components

### ملحوظة

استخدام المكونات المتداخلة كما في challemge السابقة لتقديم المكونات.

### حل

ما يلي هو الحل لل chakkenge ، حيث تجعل الحمضيات و NonCitrus في مكون ثم يتم تقديمها في آخر:

 `class Fruits extends React.Component { 
  constructor(props) { 
    super(props); 
  } 
  render() { 
    return ( 
      <div> 
        <h2>Fruits:</h2> 
        <NonCitrus /> 
        <Citrus /> 
      </div> 
    ); 
  } 
 }; 
 
 class TypesOfFood extends React.Component { 
  constructor(props) { 
     super(props); 
  } 
  render() { 
    return ( 
      <div> 
        <Fruits /> 
        <Vegetables /> 
      </div> 
    ); 
  } 
 }; 
` 

### روابط ذات صلة:

*   [المكونات والدعائم](https://reactjs.org/docs/components-and-props.html)
*   [المكونات المتداخلة](http://www.reactjstutorial.net/nested-components.html)