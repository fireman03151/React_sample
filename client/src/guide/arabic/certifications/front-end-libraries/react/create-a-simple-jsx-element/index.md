---
title: Create a Simple JSX Element
localeTitle: إنشاء عنصر بسيط JSX
---
## إنشاء عنصر بسيط JSX

هذا التحدي بسيط إلى حد ما ، واستبدل div في عنصر JSX مع h1 `javascript const JSX=<h1></h1>;` ثم أضف النص المطلوب. `javascript const JSX = <h1>Hello JSX!</h1>;`