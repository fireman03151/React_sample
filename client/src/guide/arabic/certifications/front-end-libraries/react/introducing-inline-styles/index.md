---
title: Introducing Inline Styles
localeTitle: إدخال الأنماط الداخلية
---
## إدخال الأنماط الداخلية

## حل

يمكن أن يكون هذا الأمر صعبًا قليلاً لأن JSX يشبه إلى حد بعيد HTML ولكن **ليس نفسه** .

لنستعرض الخطوات حتى تعرف الفرق. قم أولاً بتعيين علامة النمط على **كائن JavaScript** .

 `class Colorful extends React.Component { 
  render() { 
    return ( 
      <div style={{}}> 
        Big Red 
      </div> 
    ); 
  } 
 }; 
` 

الآن لديك تعيين علامة النمط الخاص بك إلى كائن فارغ. لاحظ كيف أن هناك مجموعتين من الأقواس المتعرجة. هذا اختلاف مهم بين JSX و HTML.

ثانيًا ، لنقم بتعيين اللون إلى اللون الأحمر.

 `class Colorful extends React.Component { 
  render() { 
    return ( 
      <div style={{ color: 'red' }}> 
        Big Red 
      </div> 
    ); 
  } 
 }; 
` 

أخيرًا ، لنقم بتعيين حجم الخط إلى 72 بكسل.

### المفسد

 `class Colorful extends React.Component { 
  render() { 
    return ( 
      <div style={{ color: 'red', fontSize: '72'}}> 
        Big Red 
      </div> 
    ); 
  } 
 }; 
` 

لاحظ كيف أن سمة **JSX** هي **camelCase** . هذا اختلاف مهم آخر يجب تذكره حول JSX. بالإضافة إلى ذلك ، ربما لاحظت عدم وجود وحدة. في JSX ، عند تعيين سمة fontSize ، تكون **الوحدة اختيارية** وسيتم تعيينها تلقائيًا على px إذا لم يتم تعيينها يدويًا.