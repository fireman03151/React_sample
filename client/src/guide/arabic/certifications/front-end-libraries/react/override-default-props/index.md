---
title: Override Default Props
localeTitle: تجاوز الدعائم الافتراضية
---
## تجاوز الدعائم الافتراضية

لقد تجاوزت هذا التحدي القيمة الافتراضية `quantity` الدعائية لمكون العناصر. حيث يتم تعيين القيمة الافتراضية `quantity` إلى `0` .

 `const Items = (props) => { 
  return <h1>Current Quantity of Items in Cart: {props.quantity}</h1> 
 } 
 
 Items.defaultProps = { 
  quantity: 0 
 } 
` 

لتجاوز قيمة الدعائم الافتراضية ، تكون الصيغة التي يجب اتباعها هي

 `<Component propsName={Value}/> 
` 

باتباع "بناء الجملة" ، يجب أن يتم تعريف التعليمة البرمجية التالية أدناه التعليمات البرمجية المحددة

 `<Items quantity={50}/> 
` 

سيؤدي هذا إلى تجاوز القيمة من `0` إلى `50`