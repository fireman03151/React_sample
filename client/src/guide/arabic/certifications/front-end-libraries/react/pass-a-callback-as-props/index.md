---
title: Pass a Callback as Props
localeTitle: تمرير رد الاتصال على النحو الدعائم
---
## تمرير رد الاتصال على النحو الدعائم

### وصف

*   إضافة مكون `GetInput` إلى طريقة `GetInput` في MyApp ، ثم تمريرها دعامة تسمى `inpu` t المعينة إلى `inputValue` من حالة MyApp. أيضا إنشاء دعامة تسمى `handleChange` وتمرير `handleChange` معالج `handleChange` ذلك.
*   إضافة `RenderInput` إلى طريقة `RenderInput` في MyApp ، ثم إنشاء دعم يسمى `input` وتمرير `inputValue` من حالة إليه.

### إشارة

*   `state` هي خاصية من فئة `Myapp` ، لذلك استخدم "this.state" للحصول على قيمة الكائن
*   لمعرفة المزيد عن الحالة والدعائم ، اقرأ [State and Lifecycle](https://reactjs.org/docs/state-and-lifecycle.html) and [Components and Props](https://reactjs.org/docs/components-and-props.html) .

### حل

 `class MyApp extends React.Component { 
  constructor(props) { 
    super(props); 
    this.state = { 
      inputValue: '' 
    } 
    this.handleChange = this.handleChange.bind(this); 
  } 
  handleChange(event) { 
    this.setState({ 
      inputValue: event.target.value 
    }); 
  } 
  render() { 
    return ( 
       <div> 
        { /* change code below this line */ 
        <GetInput input={this.state.inputValue} handleChange={this.handleChange}/> 
        } 
        { /* change code above this line */ 
        <RenderInput input={this.state.inputValue}/> 
        } 
       </div> 
    ); 
  } 
 }; 
`