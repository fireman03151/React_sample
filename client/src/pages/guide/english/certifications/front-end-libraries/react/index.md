---
title: React
---
## React

![React](https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/React-icon.svg/200px-React-icon.svg.png)

React or ReactJS is a front-end library which is used to create beautiful, reusable and interactive User Interfaces.

React is written in JavaScript and JSX. It is designed with reusability in mind and follows a component based pattern, which means write once and use infinitely.

It is available as a CDN for browser based development or as a npm package for Command line shell based editing.

React has been developed under the Facebook Open Source project, and is actively maintained by the Open Source community.

#### More Information:
<!-- Please add any articles you think might be helpful to read before writing the article -->
1. [React](https://reactjs.org/)
2. [React Open Source page](https://github.com/facebook/react/)
3. [React Forum](https://discuss.reactjs.org/)
