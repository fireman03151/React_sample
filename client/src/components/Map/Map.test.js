/* global expect jest */

import React from 'react';
import { useStaticQuery } from 'gatsby';
import { render } from '@testing-library/react';

import { Map } from './';
import mockChallengeNodes from '../../__mocks__/challenge-nodes';

beforeEach(() => {
  useStaticQuery.mockImplementationOnce(() => ({
    allChallengeNode: {
      nodes: mockChallengeNodes
    }
  }));
});

// set .scrollTo to avoid errors in default test environment
window.scrollTo = jest.fn();

test('<Map /> snapshot', () => {
  const { container } = render(<Map {...props} />);

  expect(container).toMatchSnapshot('Map');
});

const props = {
  forLanding: true
};
