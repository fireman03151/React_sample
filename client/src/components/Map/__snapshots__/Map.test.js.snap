// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> snapshot: Map 1`] = `
<div>
  <div
    class="map-ui"
    data-test-label="learn-curriculum-map"
  >
    <ul
      data-test-label="certifications"
    >
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-one/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-one/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-one/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-one/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-one/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-two/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-two/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-two/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-two/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-three/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          class="btn link-btn btn-lg"
          href="/learn/super-block-three/"
        >
          <div
            style="display: flex; justify-content: space-between; align-items: center;"
          >
            <div
              class="LazyLoad"
            />
          </div>
          <svg
            aria-hidden="true"
            fill="inherit"
            height="20px"
            version="1.1"
            viewBox="0 0 16 20"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <polygon
                points="-2.68014473e-15 -1.06357708e-13 2.01917516 -1.06357708e-13 8.99824941 9.00746464 2.01917516 18.0149293 -2.66453526e-15 18.0149293 7.00955027 9"
              />
              <polygon
                points="7.99971435 -1.06357708e-13 10.0188895 -1.06357708e-13 16.9979638 9.00746464 10.0188895 18.0149293 7.99971435 18.0149293 15.0092646 9"
              />
            </g>
          </svg>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
